components/comments/comment-adder.jsx

```diff
@@ -0,0 +1,64 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { postsNewComment } from "../../utils/api";
+import { FcVoicePresentation,FcApproval, } from "react-icons/fc";
+import { BiSolidMessageSquareError } from "react-icons/bi"
+
+const NewComment = ({ setComments }) => {
+  const { article_id } = useParams();
+  const [newComment, setNewComment] = useState("");
+  const [err,setErr] = useState(null)
+  const [successComment, setSuccessComment] = useState(false)
+
+  function handleNewComment(event) {
+    setNewComment(event.target.value);
+  }
+
+  function submitNewComment(event) {
+    event.preventDefault();
+postsNewComment(article_id, newComment).then((addedComment) => {
+      setNewComment("");
+      setComments((currComment) => {
+         return [addedComment, ...currComment];
+        
+      });
+      setSuccessComment(true)
+      setErr(null)
+    }).catch((err)=>{
+        setSuccessComment(false)
+        setErr("Can't comment right now, please try again later")
+        setNewComment("");
+      setComments((currComment) => {
+         return [...currComment];
+        
```

Comment:
Nice error handling for when the comment can't be posted due to internet issues 👍